### ISSUE TITLE
Helpdesk training to support new homepage Drupal elements

### ISSUE THREAD

**Issue by jilladams** [department-of-veterans-affairs/va.gov-cms#11802]
The revised homepage is going to soft launch in Q4. 
It will initially live in parallel with the current homepage, with a modal that allows users to opt in to view it.
At some point in Q1, timing TBD based on feedback, we will replace the old homepage with the new. 

At that point, there will be several pieces of content on the homepage that are editable in the CMS.

## Editable content on homepage
* 2 types of custom block - OPIA users with Homepage Manager role
* 2 entityqueues  - OPIA users with Homepage Manager role
* 2 menus - Drupal Admins

We will not be creating a dashboard to manage access / pointers to editing these items, because this will be a very small set of affected editors. Public Websites will train helpdesk, so Helpdesk can be ready to support OPIA users or content folks who come in with questions / requests for help.

This ticket represents doing the training / developing any required materials.

## Beta homepage
https://staging.va.gov/new-home-page/

### Editors
OPIA users. (Same users who can create / modify the existing [Promo block content](https://prod.cms.va.gov/admin/structure/block/block-content?info=&type=promo) that will ultimately be replaced by these new blocks, when new homepage replaces the old.)

Role required: Homepage Manager role, 

### Where it appears in FE
Two areas of the homepage are populated by custom blocks: 
![promo blocks](https://user-images.githubusercontent.com/85581471/206039352-49f94fe7-d0ec-4479-bec5-632345ac3f31.png)

To change which custom block appears on the homepage, Editors must: 
1. Create / edit a custom block
2. Edit the entityqueue for that block type to make the new block visible

### Create blocks
/admin/structure/block/block-content
Role required: Drupal Admin, or Homepage Manager (new role)

**Add custom block to create, or Filter to by Block type to see existing blocks:**
![Screen Shot 2022-12-06 at 2 13 30 PM](https://user-images.githubusercontent.com/85581471/206035019-bd877b84-8d27-4d07-9229-b6a0e9764e61.png)

**Create: Add custom > Choose block type:** 
![Screen Shot 2022-12-06 at 2 14 04 PM](https://user-images.githubusercontent.com/85581471/206035075-48bef518-7e0f-4b82-9e61-e3aafd2b6112.png)

**Block form: populate required fields**
**V2 Home page Benefit promo form:** 
![Screen Shot 2022-12-06 at 2 16 07 PM](https://user-images.githubusercontent.com/85581471/206035609-855718e5-2f72-40b3-937b-fc8e32f409b3.png)
* Block description: not shown to Veterans, just helps identify the block in the CMS list of blocks

**V2 Home page News spotlight form:** 
![Screen Shot 2022-12-06 at 2 19 47 PM](https://user-images.githubusercontent.com/85581471/206035874-eca389f8-32b8-45d2-ba37-99e5051cbecd.png)'
* Block description: not shown to Veterans, just helps identify the block in the CMS list of blocks
* Image is required and should use 1:1 ratio. (1:1 ratio is not validated.)
* Links should be functional URLs or node links. (Not fully validated.)
* CTA: currently a 1-option list, "Read full article." We may add additional options in future. 

### Modify entityqueues
/admin/structure/entityqueue

Edit the entityqueue for that block type:
  1. Remove the current promoted block
  2. Select the new custom block
  3. Add item
  4. Save the queue 
 
**Select queue > Edit:**
![Screen Shot 2022-12-06 at 2 10 23 PM](https://user-images.githubusercontent.com/85581471/206034351-8d00c726-dc34-4c90-a36f-67f19243a95f.png)

**Remove currently promoted block:**
![Screen Shot 2022-12-06 at 2 10 38 PM](https://user-images.githubusercontent.com/85581471/206034379-9ce6dcb6-ae98-484c-b298-f12af5d2fab2.png)

**Select > Add item > Save to add new block:**
![Screen Shot 2022-12-06 at 2 10 49 PM](https://user-images.githubusercontent.com/85581471/206034390-d6f71f00-1855-4391-951f-ca870d36a2b1.png)

### Editors
Drupal Admins. 
Role required: Drupal admin

### Where it appears in the FE
Three areas of the new homepage are powered by Menus
1. Top pages
2. Other search tools
3. Hub list

![Screen Shot 2022-12-06 at 2 31 03 PM](https://user-images.githubusercontent.com/85581471/206038493-3e2df94a-c5d5-49a7-b820-5f28321f81d8.png)

### Top pages
admin/structure/menu/manage/popular-on-va-gov

This menu was created as "Popular on VA.gov" but the menu name has since changed to "Top pages."

### Other Search tools
admin/structure/menu/manage/other-search-tools

### VA.gov home page benefit hub list
admin/structure/menu/manage/home-page-hub-list

<img width="569" alt="Screen Shot 2023-02-10 at 2 49 59 PM" src="https://user-images.githubusercontent.com/85581471/218216111-6813e997-97e2-49a2-bfc3-81e06a11b251.png">

The old homepage contains this Hub list, managed by content on Benefit Hub Landing Pages + an entity queue. 
The new homepage contains the same list, in a new order, now managed by a menu. 
When the new homepage hard launches, we will deprecate the old fields on BHLPs and the old entityqueue.

Links in this new menu should include selection of an Icon, and copy in the Link Summary field, which displays in the front-end below the icon and Menu link title.

## Acceptance Criteria 
TBD

**Comment by jilladams** [comment]
@dsasser FYI this exists, to train Helpdesk on how homepage works, so they can support editors post-launch. This is based on latest iteration of permissions. 

@EWashb @stefaniefgray FYI the ticket is updated with details and ready for review.

**Comment by EWashb** [comment]
@jilladams when you get a better idea of the launch can you please update us here? It will help me time this appropriately

**Comment by jilladams** [comment]
Launch timeline is maintained here in realtime as it develops: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/8877

**Comment by jilladams** [comment]
@EWashb launch timeline is updated in #8877. 
tl;dr: 

- 12/19 (sprint 74) - the path /new-home-page should go live on Prod, quietly, but available to anyone who finds it.
- 1/3 (sprint 75) - modal will launch to the VA.gov homepage (will show once per session) that makes the new page known / allows opting in to view it

**Comment by EWashb** [comment]
@stefaniefgray let's sync on this for training purposes.

**Comment by jilladams** [comment]
@stefaniefgray @EWashb Hiya - FYI that I'm in office today / tomorrow, and out Thurs/Friday. Our timing to launch the modal is very slightly TBD on a couple of bugs in progress this week, but we are still aiming for launch during week of 1/3 if possible. Let me know if you all have any questions about anything here in prep!

**Comment by EWashb** [comment]
Do either of you know how many homepage editors there are?

**Comment by jilladams** [comment]
Good question - it varies by area of the homepage, so I'll add those notes into the body of the ticket, so it's easier to grok.

**Comment by jilladams** [comment]
Added. tl;dr: 
1. Custom blocks & EntityQueues = OPIA users with Homepage Manager role. Likely to be 3-4 people to start out, but we don't have those exact people identified.
2. Menus = Drupal admins.

Also: late breaking news from Dave in a meeting I am in this minute: We will _not_ be launching the modal week of Jan 3. Dave to do some internal review with stakeholders to explain the new design, so timing is now closer to mid-Jan, TBD that feedback cycle.

**Comment by EWashb** [comment]
Thanks for the update, Jill! Want to try to meet today on it anyway? I can see if @stefaniefgray is available.

**Comment by jilladams** [comment]
Sure

**Comment by EWashb** [comment]
Most likely just KB articles for reference. Video prob wont be necessary. 3 editors most likely. May need to be a walk thru with editors but much farther down the road.

**Comment by EWashb** [comment]
@jilladams @wesrowe I want to make sure we don't drop this. Do you have an updated timeline to add to the ticket about when these KB articles should be published by?

cc: @TroyA6 @stefaniefgray

**Comment by jilladams** [comment]
Apologies - after our training, I didn't remember an ask to make KB articles. Or you're saying you all will make them and what's the necessary timeline? 

RE: timing - we still don't have any timing re: soft launch / hard launch due to Dave's January blowing up and him needing to seek some internal approvals to move forward. So: mid-Feb is probably a realistic target if possible?

**Comment by stefaniefgray** [comment]
Where to request new KB articles, per @laflannery: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/new?assignees=&labels=Knowledge+Base%2C+Needs+refining&template=knowledge-base-article.md&title=Request+for+new+or+update+to+existing+KB+article+%3A+%3Cinsert+topic+here%3E

**Comment by jilladams** [comment]
Hi folks. I drafted a KB article for this here: https://prod.cms.va.gov/managing-vagov-home-page-benefit-promo-news-spotlights

This article focuses on: 
1. Where promos appear on the home page
2. How to create content for the 2 new promo types
3. How to edit
4. How to publish to the new home page

Note: 
* **only** users with the Homepage Manager role can do these tasks, and 
* The 2 pages where the work occurs (admin block content view, and entityqueues) are **NOT** visible in any menus, for people with this role. These users will have to go directly to the URLs, and won't be able to use menus to get there. 

SO: within the content I am including the explicit URLs they'll need to use. I can't make them into links, because I get 
an error stating that links to prod cms paths are only available on the VA network, and I must use a publicly available path.

@stefaniefgray @EWashb FYI, and lemme know if you have changes. We can publish this whenever you're happy with it.

**Comment by EWashb** [comment]
@jilladams was this article reviewed and published yet? Sorry I didn't respond sooner

**Comment by jilladams** [comment]
Not as far as I know.

**Comment by stefaniefgray** [comment]
@jilladams This looks fantastic! Well done

Question -- Drupal terms within this article refer to the "homepage" while the article itself uses "Home page." Is there official guidance on which one to use? (I don't think it actually matters, I'm just curious ¯\_(ツ)_/¯ )

Article is good to go @EWashb

**Comment by EWashb** [comment]
@laflannery could you take one look at this before it's published?

**Comment by EWashb** [comment]
Otherwise, ready to publish from my end.

**Comment by laflannery** [comment]
I will review today or Monday and then publish after that

**Comment by laflannery** [comment]
@jilladams I am reviewing/editing this (per Erika's request) but there is one specific functionality thing that seems unclear:

![image](https://user-images.githubusercontent.com/106678594/219791533-21cca6a2-e0e8-4618-beef-45e7f27c4957.png)
In the screenshot above it says that if you edit a block your changes will become live in the next content release - however the next heading is about publishing promo blocks to the homepage.  From this I want to assume that this section is for publishing NEW blocks?  Or is that not quite right?  Would it be Changing which blocks display on the homepage because further down you have this:
![image](https://user-images.githubusercontent.com/106678594/219791997-0b8cf682-8e52-4a55-82ac-f9b2d81c13f5.png)


It's just that this seems confusing to me (when exactly do I have to use this entityqueue process) so I want to make sure this is clear. Thanks!

**Comment by laflannery** [comment]
@EWashb @stefaniefgray @jilladams I made quite a few edits to this - listing below for future reference:

- From Stefanie's question above, I updated all instances of "home page" to "homepage". The product name in the list is one word so that's what I went with as the source of truth
- I updated the format to match our [KB article style guidelines](https://va-gov.atlassian.net/wiki/spaces/VAGOV/pages/1771831299/Knowledge+Base+content+style) to match the rest of our articles (and what I'm trying to update all older articles to look like). Most updates were:
  - Making sure all field names are in bold instead of quotes
  - Adding a ToC and anchor links
  - Clear, concise headings
  - Limited page intro text
  - Structuring content a bit more expectedly (each section had numbered steps for example)
- Fixing URL - this is an ongoing issue and I definitely need to get it fixed as we are asking more folks to create KB articles.  Nobody knows where in the Parent link heirarchy to add things and the logical place `<CMS Knowledge Base>` actually breaks the template whereas where I placed it to fix this `CMS Knowledge Base (disabled)` is actually just fine and dandy - Shenanigans!

Updated article is still in draft of course: https://prod.cms.va.gov/help/managing-homepage-content-blocks
- I will wait to see what Jill says about my confusion on the publishing (because I updated it based on what I think might be correct) 
- And also I updated it to assume the homepage is already published since this is happening Tuesday and I figured it's close enough maybe we just wait and do it Tuesday morning.

**Comment by jilladams** [comment]
I see "Homepage" all over the KBs, so that's fine. In other places for this project, Michelle Middaugh consistently used "home page" 2 words - I don't know a definitive right/wrong, so fine to leave it this way and we can edit later as needed.

Re publishing, good call on the confusion, thank you Laura. I made these changes: 
* Under create a block > at "Save" -- "Your block must then be selected to appear on the homepage, via instructions below."
* Under edit a block > Save -- "If the block you have edited is already promoted to the new VA homepage, your changes will become visible after the next content release."

Those changes are saved, still in Draft, and if that works: Yes, fine to publish on Tuesday!

**Comment by laflannery** [comment]
Thanks @jilladams  I will publish this tomorrow

**Comment by laflannery** [comment]
Published and closed

**Comment by jilladams** [comment]
Note: since homepage launch is now planned for end of May 2023, Dave is reviewing KBs in prep for onboarding OPIA to manage homepage topics. This KB is too technical for the audience, so Dave will be drafting a few alternate KBs that are more geared toward OPIA's level of CMS usage / familiarity. We'll keep this content (maybe archived?) for Helpdesk reference if you need to support, so you have the full technical background.  FYI @stefaniefgray @TroyA6 and maybe @laflannery if you're still helping with KB article workflows. These drafts are likely to come in next week and need review before Memorial Day, so please leave some room in your sprint for that, if possible. 

Also noting: the homepage content blocks were developed with the idea that multiple blocks could be created and swapped in. For a variety of reasons, this is less desirable for the editor experience vs. just editing a single instance of the block. So: Dave's KB drafts will direct OPIA to edit the existing News promo block, vs. creating new ones and replacing existing. (I asked and he answered: this isn't a problem for content retention policies since the block content is just pointing to other content that will follow retention policies.) FYI @wesrowe
